### ISSUE TITLE
Ran into a WidevineClient error when scraping account? 

### ISSUE THREAD

**Issue by deadbeatdandylyon** [sim0n00ps/OF-DL#174]
Updating Ticket:

Ran Scrape and receiving the following error message. 

Have confirmed the latest version of OF DL V1.7.35

The below image is from lets say "Creator1" 

![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/6bdc24da-c61a-4afc-9cc7-69fcbb66b058)

Attempted to run a scrape on an another creator with DRM call them "Creator2" - although in the image you will see version 1.7.34 (this is because I didn't re-name the main folder)

This was a brand new scrape no preexisting data 

![Screenshot 2023-09-25 081929](https://github.com/sim0n00ps/OF-DL/assets/59564520/3e1f07f3-8311-49a8-b02a-ce361c1827ae)

However, when I ran a scrape on a non DRM account it worked perfectly fine.

**Comment by deadbeatdandylyon** [comment]
I attempted to run another account just to see if it was a one off, and not only did I still get the widevine error but also this as well 

![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/92fd62ba-4658-464a-bb51-8d1f7ed56f14)

**Comment by mediaburnwayne** [comment]
Getting same error, But looks like still being decrypted

**Comment by deadbeatdandylyon** [comment]
I have like maybe one or two files that are not being decrypted,

**Comment by mediaburnwayne** [comment]
Now that I check mine are not either. everything has worked flawless until the hiccup last week on OF

**Comment by deadbeatdandylyon** [comment]
@sim0n00ps suggested to create new "device_client_id_blob and device_private_key" so that's what I'm going to do and will update!

**Comment by sim0n00ps** [comment]
I will have a look into this and debug it properly

**Comment by deadbeatdandylyon** [comment]
Well I will add just so that you are aware, I did delete all OF DL files aside from the user folder, I redownloaded the zip and unzipped everything, I have also repaired both the .NET 7.0 and repaired the C++ Redistributable Packages just to be on the safe side.

**Comment by deadbeatdandylyon** [comment]
So i'm attempting to get the new keys but running into a slight issue and i'm not sure where i'm at fault

![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/52e1969d-76e4-4424-aab1-925177a86831)

**Comment by deadbeatdandylyon** [comment]
IS there a particular version of Frida I should be using?

**Comment by sim0n00ps** [comment]
You've created an x64 device, make sure it's a Pixel 6 running Android Pie

**Comment by deadbeatdandylyon** [comment]
Now the next issue that has come up when I try to run the dump_keys getting the following error 

![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/812bc876-c6ec-4264-a1e6-b57c3a08f25d)

**Comment by sim0n00ps** [comment]
Thanks for all of the updates @deadbeatdandylyon I really appreciate it, great info for anyone else who has the issue down the line. I've been letting it rip on some profiles with DRM and the first time round I got the errors. But I haven't seen it again after running 2-3 more scrapes.

**Comment by deadbeatdandylyon** [comment]
You're welcome, I know it's not all neat of a reply but I hope it makes sense. I'm going to run it on one more creator just to also double check it's not a fluke but I think things should be good to go.

**Comment by deadbeatdandylyon** [comment]
Okay so back to the same issue again. I ran the original account I was having issues with and got the same errors. 

I deleted the metadata folder and everything. 

![Screenshot 2023-09-25 155225](https://github.com/sim0n00ps/OF-DL/assets/59564520/07ce72e0-67a8-41c6-b568-1e348d09bb31)

**Comment by deadbeatdandylyon** [comment]
I wonder if it would be that it's coming up as a M4A file be the problem? 

Have you come across this before?

**Comment by sim0n00ps** [comment]
Yes an MP4 and M4A file should be downloaded, decrypted and merged into 1 MP4

**Comment by deadbeatdandylyon** [comment]
because based off that screen shot the file in question that it can't find is the mp4 however I see two versions of it an M4A and MP4

![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/607a349c-f5bf-4bdb-a9ad-4c19964e6e31)

**Comment by sim0n00ps** [comment]
Those 2 files are encrypted and cannot be played. The reason you get the cannot find file error is because the decrypted version is never made because you don't have the decryption key which links to the main exception about it not being able to parse the licence.

**Comment by deadbeatdandylyon** [comment]
Ahh gotcha, I mean granted it's only a 5 second video, and it was decrypted previously when I did the scrape and so far everything else seems to be decrypting, but is this even anything that you are able to resolve or is this just a wait and see and hopefully it will go away on it's own type of thing, i know you mentioned you will debug it but just trying to educate myself a little bit on it

**Comment by sim0n00ps** [comment]
The code is set up correctly, I have a working CDM, OF haven't changed anything, when running and debugging the code the process seems fine + requests are being made correctly and decryption keys being returned without any issue. It probably is one of those things that you hope fixes itself.

**Comment by deadbeatdandylyon** [comment]
Okie dokie, I will keep an eye out on everything and see how it goes. Now it's up to 3 videos that failed, hopefully things will fix itself on it's own then, thank you for the information

**Comment by deadbeatdandylyon** [comment]
Received some new error messages, some the same and some different

![Screenshot 2023-09-26 005707](https://github.com/sim0n00ps/OF-DL/assets/59564520/a4ba8903-2d84-4777-bd92-d2c665b1cc5e)


![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/bd6f0827-4e44-4e5a-a474-bc3fe1795af2)

**Comment by JC02023** [comment]
my small tiny brain can not comprehend what u guys are talking about but i have the same error :(

**Comment by sim0n00ps** [comment]
> Received some new error messages, some the same and some different
> 
> ![Screenshot 2023-09-26 005707](https://github.com/sim0n00ps/OF-DL/assets/59564520/a4ba8903-2d84-4777-bd92-d2c665b1cc5e)
> 
> 
> ![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/bd6f0827-4e44-4e5a-a474-bc3fe1795af2)
> 

So just to confirm did you dump an Android 9 Pie CDM?

**Comment by deadbeatdandylyon** [comment]
That is correct I believe 

![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/edac6383-7d81-40e3-a373-bf033929c06e)

**Comment by sim0n00ps** [comment]
Yeah looks good, did you use the same device you used the first time round or is that a fresh device you made yesterday?

**Comment by deadbeatdandylyon** [comment]
I made a completely new device as I had to reinstall the software to begin with, I deleted the old keys and moved over the new keys and re pathed them into the directory of `cdm/devices/chrome_1610`

![image](https://github.com/sim0n00ps/OF-DL/assets/59564520/e0c63c1c-04e0-4d97-a760-eba6656c0806)

**Comment by sim0n00ps** [comment]
I'm not sure then, it might mean you will have to try and get a CDM from a newer version of Android even though mine worked fine last night.

**Comment by deadbeatdandylyon** [comment]
Well what's interesting is while I'm waiting for this issue to clear up. There was another OF scraper I found and with using the same keys i just copied them and pasted them over to this other OF Scraper and I have scraped at least 5 profiles without any issues and everything is decrypting and downloading without any complications. 

So that being said I know your program works, I know their program works, I know the keys work. 

So my only other option that I can try is a complete erase of everything and start over with either an OS wipe or VMware test.

but these are just my findings thus far.

**Comment by deadbeatdandylyon** [comment]
Which Frida Version are you currently using with your CDM 16.0.4 or the new 16.1.4, because the one I'm using is 16.1.4, I might be curious if I went back to the version that your using or an older version that might be an issue as well. As well as updated versions of Bento / Yt / and I "think" ffmpeg 

so these are other thoughts as well that could possibly playing into the problem that i'm having

**Comment by sim0n00ps** [comment]
I'm fairly sure I use 16.1.3 I'm not at my PC atm so I can't check for sure but I haven't changed it at all. You also have to have the same version of Frida installed via pip as well so they both have to match.

**Comment by deadbeatdandylyon** [comment]
Well i'm not sure how to verify the PIP

But I do know I was having issues with the `dump_keys` 

When I was trying to run standard `python dump_keys.py` 

I did have an issue / so I ran this `pip3 install -r requirements.txt` then wound up having an issue that I had to "downgrade the protobuf package to 3.20.x or lower" 

So I had to find a resolution to that and found this [https://github.com/wvdumper/dumper/issues/59](url) which stated to run `pip install protobuf==3.20.3` so that's what I did

So not sure if having version Frida 16.1.4 and then this version of PIP if that's where the problem is or if they are even related not sure

**Comment by sim0n00ps** [comment]
Yes that is correct. Run pip show frida and see what version you have installed.

**Comment by deadbeatdandylyon** [comment]
Yes Is currently showing Version 16.1.4

![Screenshot 2023-09-26 050403](https://github.com/sim0n00ps/OF-DL/assets/59564520/fbd1e505-d640-4996-9f7e-94a52d7c2cc6)

**Comment by sim0n00ps** [comment]
Okay maybe try 16.1.3 and keep downgrading if you need to.

**Comment by deadbeatdandylyon** [comment]
Alright I'll give that a go and see how many versions I have to go back. and will update once I have more information

**Comment by deadbeatdandylyon** [comment]
Well update I went down to 16.1.3 and came across the same issue. i'll work on it more at a later time but going down one by one is a bit of tedious work wish there was a more quicker approach

**Comment by sim0n00ps** [comment]
I can add support for some other ways to get the decryption keys like keysdb.net which will bypass the problem if you cannot fix it

**Comment by deadbeatdandylyon** [comment]
if this is a stupid question, i'm sorry

is there any way that you can add the keys that lets say you have and already add them to the CDM folder so that they are there upon unzipping or are the keys unique to the persons computer 

just curious 

i'll keep you posted on everything if it comes to that i'll let you know but i don't think it should.

**Comment by deadbeatdandylyon** [comment]
> > if this is a stupid question, i'm sorry
> > is there any way that you can add the keys that lets say you have and already add them to the CDM folder so that they are there upon unzipping or are the keys unique to the persons computer
> > just curious
> > i'll keep you posted on everything if it comes to that i'll let you know but i don't think it should.
> 
> All keys are different and I recommend you make your own so they don't get revoked due to other people using them. I can generate you 2 new ones and send you them on simpcity when I get a chance and see if that makes a difference?

I'll see what I can do from my end. if i get to my wits end then i'll let you know,

**Comment by sim0n00ps** [comment]
> > > if this is a stupid question, i'm sorry
> > > is there any way that you can add the keys that lets say you have and already add them to the CDM folder so that they are there upon unzipping or are the keys unique to the persons computer
> > > just curious
> > > i'll keep you posted on everything if it comes to that i'll let you know but i don't think it should.
> > 
> > 
> > All keys are different and I recommend you make your own so they don't get revoked due to other people using them. I can generate you 2 new ones and send you them on simpcity when I get a chance and see if that makes a difference?
> 
> I'll see what I can do from my end. if i get to my wits end then i'll let you know,

I've sent you some new keys on simpcity

**Comment by deadbeatdandylyon** [comment]
Still getting the Widevine error and everything else.

**Comment by superfan12345** [comment]
OF prob changed something

**Comment by sim0n00ps** [comment]
As far as I'm aware OF haven't changed anything at all.

**Comment by superfan12345** [comment]
it seems odd that it would just stop working. as i used it days before on the same creator and it worked. so something must have changed either of or something we use to use the rip.

**Comment by sim0n00ps** [comment]
Well I also used it 2 days ago and I just saw it once or twice. The rest of the time it was fine and worked perfectly. It downloaded the MP4 and M4A, got the decryption key and decrypted the files and merged them together.

**Comment by sim0n00ps** [comment]
I've downloaded some DRM videos tonight without any issue so I'm going to close this now and just come to the conclusion it was an issue on OF's side

**Comment by BigGabagool** [comment]
> I made a completely new device as I had to reinstall the software to begin with, I deleted the old keys and moved over the new keys and re pathed them into the directory of `cdm/devices/chrome_1610`
> 
> ![image](https://user-images.githubusercontent.com/59564520/270588104-e0c63c1c-04e0-4d97-a760-eba6656c0806.png)

Hi, sorry to reply to a closed issue.

I dont know if you've fixed your problem or not, but I had a similar problem where files were being downloaded in two parts even with the blob and key files being "correct". Was up all night restarting the file making step.

Tried again this morning and did something different, I closed the chrome tab then closed the chrome app after starting the video and it seemed to do the trick.

Sorry if this doesn't fix your issue, I'm not savvy on these kind of things.
